Upgrade list-manager

lb803 opened the pull request:
This upgrades list-manager to include the following new commits:

 - [add integration tests for behave](https://github.com/lb803/list-manager-skill/commit/633645e0ba4fbda0bd77521fa608462a33635e74)
 - [Merge pull request #13 from Mycroft-eus/basque](https://github.com/lb803/list-manager-skill/commit/3d50fe4383ebddccda1cf4983ab9c5358a1bcd67)
 - [fix issue related to handle_add.1.intent.json](https://github.com/lb803/list-manager-skill/commit/0820c4866892ed6adeacc347f2502f715491deaa)
 - [Merge tag 'v0.0.2' into develop](https://github.com/lb803/list-manager-skill/commit/681c0270f484e41d1e1d8e1be2acf352539ce46d)
 - [Merge branch 'release/0.0.2'](https://github.com/lb803/list-manager-skill/commit/18ba25fe76a43df9a746095c6f3b7217b90bc7e0)
 - [make list.name.rx less greedy](https://github.com/lb803/list-manager-skill/commit/41fed312395e023bda9e2fd8f3a576375164bdfa)
 - ["Changes for Basque language"](https://github.com/lb803/list-manager-skill/commit/06135e914bd6f60bddd6b65abd6c82bf89787b1f)
 - [add test for item.already.exists.dialog](https://github.com/lb803/list-manager-skill/commit/67ea6f13097163a0f38539204870cbc317dd39a0)
 - [rename item.already.exists.dialog for consistency](https://github.com/lb803/list-manager-skill/commit/5dfbc2cb3e44b3e4840c18ac04b811440196b544)
 - [cleanup](https://github.com/lb803/list-manager-skill/commit/2f200ce718ea416b7ccb37d9fd62bb12729a7700)
 - [replace string() method with join_list()](https://github.com/lb803/list-manager-skill/commit/0571c27e2ce144b79c3559b2adc13d7bb16db1ad)
 - [Merge branch 'feature/streamline-conj-and' into develop](https://github.com/lb803/list-manager-skill/commit/9b384c2a24ddfa8350dcd3181d9911e5f1ec0902)
 - [rename transtlation files](https://github.com/lb803/list-manager-skill/commit/07b99d1a457cd4854a9c8f4679f9dcc5df971997)
 - [turn conj.value to and.dialog](https://github.com/lb803/list-manager-skill/commit/d50d2601e08d0bb4486f50107cd2ce6da8764c39)
 - [Merge pull request #11 from lb803/pull10](https://github.com/lb803/list-manager-skill/commit/dfffa2365a335cb4d49b3cfd080cc2665bb9d5a2)
 - [consistency](https://github.com/lb803/list-manager-skill/commit/18849bad1182ca8591c52047a75866f199e853cd)
 - [Merge pull request #5 from devs-mycroft/translations](https://github.com/lb803/list-manager-skill/commit/d4750c743f97cd663caa0954f459789235c49487)
 - [Added dialog if item already exists](https://github.com/lb803/list-manager-skill/commit/924d608263343a9e3f6308d461a56a78be7cd2fd)
 - [Correctly checks if a list is empty before reading out the contents. Checks if an item is on the list before adding it.](https://github.com/lb803/list-manager-skill/commit/480f223761221f37a195517157aec8207f789e82)
 - [Update translations](https://github.com/lb803/list-manager-skill/commit/b3df69d7384995261750532dee581e1f0b36ebad)
 - [Added icon](https://github.com/lb803/list-manager-skill/commit/64f2702c90e8ff7e43e3c79e9a2c54234b8f0895)

<sub>Created with [mycroft-skills-kit](https://github.com/mycroftai/mycroft-skills-kit) v0.3.16</sub>

devops-mycroft commented:
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/skills/PR-1541)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/skills/PR-1541/logs/skills.log), [audio.log](https://reports.mycroft.ai/skills/PR-1541/logs/audio.log), [voice.log](https://reports.mycroft.ai/skills/PR-1541/logs/voice.log), [bus.log](https://reports.mycroft.ai/skills/PR-1541/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/skills/PR-1541/logs/enclosure.log)

lb803 commented:
Hi there!
I understand this is an internal error? Should I wait before I try upgrading the skill again?

forslund commented:
Yes seems there is need for some fixing here before the tests can be run. We can re-trigger this job once the skill tester is fixed.

lb803 commented:
Thank you!
Then I'll check if something needs to be changed skill-wise.

forslund commented:
Right, my bad. you'll likely need to redo the msk upgrade command to get it based of the latest 21.02 once the fix (#1542 or some equivalent fix) is merged. but I can ping you when it's ready.

devops-mycroft commented:
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/skills/PR-1541)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/skills/PR-1541/logs/skills.log), [audio.log](https://reports.mycroft.ai/skills/PR-1541/logs/audio.log), [voice.log](https://reports.mycroft.ai/skills/PR-1541/logs/voice.log), [bus.log](https://reports.mycroft.ai/skills/PR-1541/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/skills/PR-1541/logs/enclosure.log)

forslund commented:
Seems like @krisgesling could get this to use the updated test framework. Now it has run the tests but there are some failures for your skill it seems.

lb803 commented:
@forslund Thank you for this!

I need a little guidance:

1. How do I push new commits to the `upgrade-list-manager` branch on my `mycroft-skills` fork? I understand skills are reported as submodules, but if I run `git submodule update` nothing magic happens.
2. How do I test skills locally to get the same results as your CI software? When I run `python -m test.integrationtests.skills.runner /opt/mycroft/skills/llist-manager-skill/` I get a rather different result (please note one of the errors has been fixed)
```
Searching for tests under: /opt/mycroft/skills/list-manager-skill/test/intent/*.json
Event:  padatious:register_intent
Event:  padatious:register_entity
Event:  detach_intent
Event:  detach_skill
Event:  mycroft.skills.initialized
Event:  register_vocab
Event:  register_intent
Event:  recognizer_loop:utterance
Event:  detach_intent
Event:  detach_skill
Event:  add_context
Event:  remove_context
Event:  clear_context
Event:  mycroft.speech.recognition.unknown
Event:  mycroft.skills.loaded
Event:  active_skill_request
Event:  intent.service.intent.get
Event:  intent.service.skills.get
Event:  intent.service.active_skills.get
Event:  intent.service.adapt.get
Event:  intent.service.adapt.manifest.get
Event:  intent.service.adapt.vocab.manifest.get
Event:  intent.service.padatious.get
Event:  intent.service.padatious.manifest.get
Event:  intent.service.padatious.entities.manifest.get
Event:  mycroft.skills.fallback
Event:  skill.converse.request
Event:  mycroft.skill.enable_intent
Event:  mycroft.skill.disable_intent
Event:  mycroft.skill.set_cross_context
Event:  mycroft.skill.remove_cross_context
Event:  mycroft.skills.settings.changed
Event:  test-list-manager-skill.set
Event:  test-list-manager-skill:add
Event:  test-list-manager-skill:read
Event:  test-list-manager-skill:del
2021-07-29 15:49:27.609 | INFO     |  1350 | mycroft.skills.intent_services.padatious_service:train:81 | Training... (single_thread=True)
2021-07-29 15:49:27.611 | INFO     |  1350 | mycroft.skills.intent_services.padatious_service:train:83 | Training complete.

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_add.0.intent.json
Test: {
    "expected_dialog": "add.list",
    "intent": {
        "add": "create",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "add",
    "utterance": "Create a new list called films"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'add'], ['endsWith', '__type__', 'add']], ['equal', 'add', 'create'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', 'Your new list .* has been added.']]]
UTTERANCE: Create a new list called films

Evaluating message:  {'intent_type': 'test-list-manager-skill:add', 'target': None, 'confidence': 0.4166666666666667, '__tags__': [{'match': 'create', 'key': 'create', 'start_token': 0, 'entities': [{'key': 'create', 'match': 'create', 'data': [('create', 'test_list_manager_skilladd')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 3, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 3, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}], 'utterance': 'Create a new list called films', 'add': 'create', 'list': 'list', 'list_name': 'films', '__type__': 'add'}

Evaluating message:  {'name': 'ListManager.handle_add', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'Your new list films has been added.', 'expect_response': False, 'meta': {'dialog': 'add.list', 'data': {'list_name': 'films', 'item_name': None}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_add.1.intent.json
Test: {
    "expected_dialog": "add.item",
    "intent": {
        "add": "add",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "add",
    "utterance": "Add office space to my films list"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'add'], ['endsWith', '__type__', 'add']], ['equal', 'add', 'add'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', '.* added to your .* list.']]]
UTTERANCE: Add office space to my films list

Evaluating message:  {'intent_type': 'test-list-manager-skill:add', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'add', 'key': 'add', 'start_token': 0, 'entities': [{'key': 'add', 'match': 'add', 'data': [('add', 'test_list_manager_skilladd')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Add office space to my films list', 'add': 'add', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'add'}

Evaluating message:  {'name': 'ListManager.handle_add', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'office space added to your films list.', 'expect_response': False, 'meta': {'dialog': 'add.item', 'data': {'list_name': 'films', 'item_name': 'office space'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_add.2.intent.json
Test: {
    "expected_dialog": "list.found",
    "intent": {
        "add": "create",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "add",
    "utterance": "Create a new list called films"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'add'], ['endsWith', '__type__', 'add']], ['equal', 'add', 'create'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', 'You alredy have a list called .*.']]]
UTTERANCE: Create a new list called films

Evaluating message:  {'intent_type': 'test-list-manager-skill:add', 'target': None, 'confidence': 0.4166666666666667, '__tags__': [{'match': 'create', 'key': 'create', 'start_token': 0, 'entities': [{'key': 'create', 'match': 'create', 'data': [('create', 'test_list_manager_skilladd')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 3, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 3, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}], 'utterance': 'Create a new list called films', 'add': 'create', 'list': 'list', 'list_name': 'films', '__type__': 'add'}

Evaluating message:  {'name': 'ListManager.handle_add', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'You alredy have a list called films.', 'expect_response': False, 'meta': {'dialog': 'list.found', 'data': {'list_name': 'films', 'item_name': None}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_add.3.intent.json
Test: {
    "expected_dialog": "list.not.found",
    "intent": {
        "add": "add",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "add",
    "utterance": "Add office space to my films list"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'add'], ['endsWith', '__type__', 'add']], ['equal', 'add', 'add'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', "You don't seem to have a list called .*."], ['match', 'utterance', "I couldn't find .* among your lists."]]]
UTTERANCE: Add office space to my films list

Evaluating message:  {'intent_type': 'test-list-manager-skill:add', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'add', 'key': 'add', 'start_token': 0, 'entities': [{'key': 'add', 'match': 'add', 'data': [('add', 'test_list_manager_skilladd')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Add office space to my films list', 'add': 'add', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'add'}

Evaluating message:  {'name': 'ListManager.handle_add', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': "I couldn't find films among your lists.", 'expect_response': False, 'meta': {'dialog': 'list.not.found', 'data': {'list_name': 'films', 'item_name': 'office space'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_add.4.intent.json
Test: {
    "expected_dialog": "item.already.exists",
    "intent": {
        "add": "add",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "add",
    "utterance": "Add office space to my films list"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'add'], ['endsWith', '__type__', 'add']], ['equal', 'add', 'add'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', '.* is already on your .* list.']]]
UTTERANCE: Add office space to my films list

Evaluating message:  {'intent_type': 'test-list-manager-skill:add', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'add', 'key': 'add', 'start_token': 0, 'entities': [{'key': 'add', 'match': 'add', 'data': [('add', 'test_list_manager_skilladd')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Add office space to my films list', 'add': 'add', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'add'}

Evaluating message:  {'name': 'ListManager.handle_add', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'office space is already on your films list.', 'expect_response': False, 'meta': {'dialog': 'item.already.exists', 'data': {'list_name': 'films', 'item_name': 'office space'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.0.intent.json
Test: {
    "expected_dialog": "del.list",
    "intent": {
        "del": "remove",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove my films list",
    "responses": [
        "yes"
    ],
    "evaluation_timeout": "10"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', 'Your list .* has been removed.']]]
UTTERANCE: Remove my films list
>> Are you sure you want to delete films?
SENDING RESPONSE: yes

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4166666666666667, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 3, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 3, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 2, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 2, 'from_context': False}], 'utterance': 'Remove my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'Are you sure you want to delete films?', 'expect_response': False, 'meta': {'skill': 'ListManager'}, '__type__': 'speak'}

Evaluating message:  {'utterance': 'Your list films has been removed.', 'expect_response': False, 'meta': {'dialog': 'del.list', 'data': {'list_name': 'films', 'item_name': None}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.1.intent.json
Test: {
    "expected_dialog": "del.item",
    "intent": {
        "del": "remove",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove office space from my films list",
    "responses": [
        "yes"
    ],
    "evaluation_timeout": "10"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', '.* removed from your .* list.']]]
UTTERANCE: Remove office space from my films list
>> Are you sure you want to delete office space?
SENDING RESPONSE: yes

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Remove office space from my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'Are you sure you want to delete office space?', 'expect_response': False, 'meta': {'skill': 'ListManager'}, '__type__': 'speak'}

Evaluating message:  {'utterance': 'office space removed from your films list.', 'expect_response': False, 'meta': {'dialog': 'del.item', 'data': {'list_name': 'films', 'item_name': 'office space'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.2.intent.json
Test: {
    "expected_dialog": "list.not.found",
    "intent": {
        "del": "remove",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove my films list"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', "You don't seem to have a list called .*."], ['match', 'utterance', "I couldn't find .* among your lists."]]]
UTTERANCE: Remove my films list

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4166666666666667, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 3, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 3, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 2, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 2, 'from_context': False}], 'utterance': 'Remove my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': "I couldn't find films among your lists.", 'expect_response': False, 'meta': {'dialog': 'list.not.found', 'data': {'list_name': 'films', 'item_name': None}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.3.intent.json
Test: {
    "expected_dialog": "cancelled",
    "intent": {
        "del": "remove",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove my films list",
    "responses": [
        "no"
    ],
    "evaluation_timeout": "10"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', 'Cancelled.']]]
UTTERANCE: Remove my films list
>> Are you sure you want to delete films?
SENDING RESPONSE: no

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4166666666666667, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 3, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 3, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 2, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 2, 'from_context': False}], 'utterance': 'Remove my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'Are you sure you want to delete films?', 'expect_response': False, 'meta': {'skill': 'ListManager'}, '__type__': 'speak'}

Evaluating message:  {'utterance': 'Cancelled.', 'expect_response': False, 'meta': {'dialog': 'cancelled', 'data': {}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.4.0.intent.json
Test: {
    "expected_dialog": "item.not.found",
    "intent": {
        "del": "remove",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove office space from my films list"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', '.* is not on your .* list.']]]
UTTERANCE: Remove office space from my films list

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Remove office space from my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'office space is not on your films list.', 'expect_response': False, 'meta': {'dialog': 'item.not.found', 'data': {'list_name': 'films', 'item_name': 'office space'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.4.1.intent.json
Test: {
    "expected_dialog": "item.not.found",
    "intent": {
        "del": "remove",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove office space from my films list"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', '.* is not on your .* list.']]]
UTTERANCE: Remove office space from my films list

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Remove office space from my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'office space is not on your films list.', 'expect_response': False, 'meta': {'dialog': 'item.not.found', 'data': {'list_name': 'films', 'item_name': 'office space'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.4.2.intent.json
Test: {
    "expected_dialog": "item.not.found",
    "intent": {
        "del": "remove",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove office space from my films list"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', '.* is not on your .* list.']]]
UTTERANCE: Remove office space from my films list

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Remove office space from my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'office space is not on your films list.', 'expect_response': False, 'meta': {'dialog': 'item.not.found', 'data': {'list_name': 'films', 'item_name': 'office space'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_del.5.intent.json
Test: {
    "expected_dialog": "cancelled",
    "intent": {
        "del": "remove",
        "item_name": "office space",
        "list": "list",
        "list_name": "films"
    },
    "intent_type": "del",
    "utterance": "Remove office space from my films list",
    "responses": [
        "no"
    ],
    "evaluation_timeout": "10"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'del'], ['endsWith', '__type__', 'del']], ['equal', 'del', 'remove'], ['equal', 'item_name', 'office space'], ['equal', 'list', 'list'], ['equal', 'list_name', 'films']], ['or', ['match', 'utterance', 'Cancelled.']]]
UTTERANCE: Remove office space from my films list
>> Are you sure you want to delete office space?
SENDING RESPONSE: no

Evaluating message:  {'intent_type': 'test-list-manager-skill:del', 'target': None, 'confidence': 0.4375, '__tags__': [{'match': 'remove', 'key': 'remove', 'start_token': 0, 'entities': [{'key': 'remove', 'match': 'remove', 'data': [('remove', 'test_list_manager_skilldel')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 6, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 6, 'from_context': False}, {'match': 'films', 'key': 'films', 'start_token': 5, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'end_token': 5, 'from_context': False}, {'start_token': 1, 'entities': [{'key': 'office space', 'match': 'office space', 'data': [('office space', 'test_list_manager_skillitem_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 2, 'match': 'office space', 'key': 'office space', 'from_context': False}], 'utterance': 'Remove office space from my films list', 'del': 'remove', 'list': 'list', 'list_name': 'films', 'item_name': 'office space', '__type__': 'del'}

Evaluating message:  {'name': 'ListManager.handle_del', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'Are you sure you want to delete office space?', 'expect_response': False, 'meta': {'skill': 'ListManager'}, '__type__': 'speak'}

Evaluating message:  {'utterance': 'Cancelled.', 'expect_response': False, 'meta': {'dialog': 'cancelled', 'data': {}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_read.0.intent.json
Test: {
    "expected_dialog": "read.lists",
    "utterance": "What lists do I have?",
    "intent": {
        "list": "lists",
        "what": "what"
    },
    "intent_type": "read"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'read'], ['endsWith', '__type__', 'read']], ['equal', 'list', 'lists'], ['equal', 'what', 'what']], ['or', ['match', 'utterance', 'Currently you have .*.']]]
UTTERANCE: What lists do I have?

Evaluating message:  {'intent_type': 'test-list-manager-skill:read', 'target': None, 'confidence': 1.0, '__tags__': [{'match': 'what', 'key': 'what', 'start_token': 0, 'entities': [{'key': 'what', 'match': 'what', 'data': [('what', 'test_list_manager_skillwhat')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'lists', 'key': 'lists', 'start_token': 1, 'entities': [{'key': 'lists', 'match': 'lists', 'data': [('lists', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 1, 'from_context': False}], 'utterance': 'What lists do I have?', 'what': 'what', 'list': 'lists', '__type__': 'read'}

Evaluating message:  {'name': 'ListManager.handle_read', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'Currently you have films and readings lists.', 'expect_response': False, 'meta': {'dialog': 'read.lists', 'data': {'list_name': None, 'lists_names': 'films and readings', 'list': 'lists'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_read.1.intent.json
Test: {
    "expected_dialog": "read.items",
    "intent": {
        "list": "list",
        "list_name": "films",
        "what": "what"
    },
    "intent_type": "read",
    "utterance": "What's on my films list?"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'read'], ['endsWith', '__type__', 'read']], ['equal', 'list', 'list'], ['equal', 'list_name', 'films'], ['equal', 'what', 'what']], ['or', ['match', 'utterance', 'In your .* list you have .*']]]
UTTERANCE: What's on my films list?

Evaluating message:  {'intent_type': 'test-list-manager-skill:read', 'target': None, 'confidence': 1.0, '__tags__': [{'match': 'what', 'key': 'what', 'start_token': 0, 'entities': [{'key': 'what', 'match': 'what', 'data': [('what', 'test_list_manager_skillwhat')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 5, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 5, 'from_context': False}, {'start_token': 4, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 4, 'match': 'films', 'key': 'films', 'from_context': False}], 'utterance': "What's on my films list?", 'what': 'what', 'list': 'list', 'list_name': 'films', '__type__': 'read'}

Evaluating message:  {'name': 'ListManager.handle_read', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'In your films list you have office space and snowden', 'expect_response': False, 'meta': {'dialog': 'read.items', 'data': {'list_name': 'films', 'items': 'office space and snowden'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_read.2.intent.json
Test: {
    "expected_dialog": "no.lists",
    "utterance": "What lists do I have?",
    "intent": {
        "list": "lists",
        "what": "what"
    },
    "intent_type": "read"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'read'], ['endsWith', '__type__', 'read']], ['equal', 'list', 'lists'], ['equal', 'what', 'what']], ['or', ['match', 'utterance', "You don't have any lists yet."]]]
UTTERANCE: What lists do I have?

Evaluating message:  {'intent_type': 'test-list-manager-skill:read', 'target': None, 'confidence': 1.0, '__tags__': [{'match': 'what', 'key': 'what', 'start_token': 0, 'entities': [{'key': 'what', 'match': 'what', 'data': [('what', 'test_list_manager_skillwhat')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'lists', 'key': 'lists', 'start_token': 1, 'entities': [{'key': 'lists', 'match': 'lists', 'data': [('lists', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 1, 'from_context': False}], 'utterance': 'What lists do I have?', 'what': 'what', 'list': 'lists', '__type__': 'read'}

Evaluating message:  {'name': 'ListManager.handle_read', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': "You don't have any lists yet.", 'expect_response': False, 'meta': {'dialog': 'no.lists', 'data': {}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_read.3.intent.json
Test: {
    "expected_dialog": "no.items",
    "intent": {
        "list": "list",
        "list_name": "films",
        "what": "what"
    },
    "intent_type": "read",
    "utterance": "What's on my films list?"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'read'], ['endsWith', '__type__', 'read']], ['equal', 'list', 'list'], ['equal', 'list_name', 'films'], ['equal', 'what', 'what']], ['or', ['match', 'utterance', 'Your .* list is empty.']]]
UTTERANCE: What's on my films list?

Evaluating message:  {'intent_type': 'test-list-manager-skill:read', 'target': None, 'confidence': 1.0, '__tags__': [{'match': 'what', 'key': 'what', 'start_token': 0, 'entities': [{'key': 'what', 'match': 'what', 'data': [('what', 'test_list_manager_skillwhat')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 5, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 5, 'from_context': False}, {'start_token': 4, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 4, 'match': 'films', 'key': 'films', 'from_context': False}], 'utterance': "What's on my films list?", 'what': 'what', 'list': 'list', 'list_name': 'films', '__type__': 'read'}

Evaluating message:  {'name': 'ListManager.handle_read', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': 'Your films list is empty.', 'expect_response': False, 'meta': {'dialog': 'no.items', 'data': {'list_name': 'films'}, 'skill': 'ListManager'}, '__type__': 'speak'}

==================== RUNNING TEST ====================
Test file:  /opt/mycroft/skills/list-manager-skill/test/intent/handle_read.4.intent.json
Test: {
    "expected_dialog": "list.not.found",
    "intent": {
        "list": "list",
        "list_name": "films",
        "what": "what"
    },
    "intent_type": "read",
    "utterance": "What's on my films list?"
}
Rule created  [['and', ['or', ['endsWith', 'intent_type', 'read'], ['endsWith', '__type__', 'read']], ['equal', 'list', 'list'], ['equal', 'list_name', 'films'], ['equal', 'what', 'what']], ['or', ['match', 'utterance', "You don't seem to have a list called .*."], ['match', 'utterance', "I couldn't find .* among your lists."]]]
UTTERANCE: What's on my films list?

Evaluating message:  {'intent_type': 'test-list-manager-skill:read', 'target': None, 'confidence': 1.0, '__tags__': [{'match': 'what', 'key': 'what', 'start_token': 0, 'entities': [{'key': 'what', 'match': 'what', 'data': [('what', 'test_list_manager_skillwhat')], 'confidence': 1.0}], 'end_token': 0, 'from_context': False}, {'match': 'list', 'key': 'list', 'start_token': 5, 'entities': [{'key': 'list', 'match': 'list', 'data': [('list', 'test_list_manager_skilllist')], 'confidence': 1.0}], 'end_token': 5, 'from_context': False}, {'start_token': 4, 'entities': [{'key': 'films', 'match': 'films', 'data': [('films', 'test_list_manager_skilllist_name')], 'confidence': 0.5}], 'confidence': 0.5, 'end_token': 4, 'match': 'films', 'key': 'films', 'from_context': False}], 'utterance': "What's on my films list?", 'what': 'what', 'list': 'list', 'list_name': 'films', '__type__': 'read'}

Evaluating message:  {'name': 'ListManager.handle_read', '__type__': 'mycroft.skill.handler.start'}

Evaluating message:  {'utterance': "You don't seem to have a list called films.", 'expect_response': False, 'meta': {'dialog': 'list.not.found', 'data': {'list_name': 'films'}, 'skill': 'ListManager'}, '__type__': 'speak'}

..................
----------------------------------------------------------------------
Ran 18 tests in 0.487s

OK
```

forslund commented:
The easiest way to update the PR here is to use MSK

1. Make changes to your skill
2. Push them to github
3. run `msk submit /path/to/skill`

If there is no existing PR it will create one, if there is a PR it will be updated with the new commits.


To run the tests it's easiest to use the script in the /bin folder:

from the mycroft-core folder
```
./bin/mycroft-skill-testrunner vktest -t list-manager
```

(Make Mycroft is running before starting this command)

(Your command is equivalent with above if `vktest` is left out of the command line. will trigger the Old skill test framework)

The docs are available here:
https://mycroft-ai.gitbook.io/docs/skill-development/voight-kampff/test-runner

lb803 commented:
Thank you Ake!!

krisgesling commented:
Hey just following up from Chat. 

The new system doesn't yet have mocking available, so it's grabbing your old tests but failing because the mocked values aren't there. Because it writes them out in the new form, it was pretty quick to grab them and make a slimmed down version that will work. If you put the following under `test/behave/list-manager.feature` the Voight Kampff test suite will use it instead of the old test files. 

```
Feature: list-manager

  Scenario: Create a new list
    Given an english speaking user
    When the user says "Create a new list called films"
    Then "list-manager" should reply with dialog from "add.list.dialog"

  Scenario: Add an item to an existing list
    Given an english speaking user
    When the user says "Add office space to my films list"
    Then "list-manager" should reply with dialog from "add.item.dialog"

  Scenario: Do not create list if it already exists
    Given an english speaking user
    When the user says "Create a new list called films"
    Then "list-manager" should reply with dialog from "list.found.dialog"

  Scenario: Read out the existing lists
    Given an english speaking user
    When the user says "What lists do I have?"
    Then "list-manager" should reply with dialog from "read.lists.dialog"

  Scenario: Read items on a specific list
    Given an english speaking user
    When the user says "What's on my films list?"
    Then "list-manager" should reply with dialog from "read.items.dialog"

  Scenario: Delete a specific item from a list
    Given an english speaking user
    When the user says "Remove office space from my films list"
    Then "list-manager" should reply with dialog from "confirm.deletion.dialog"
    And the user replies with "yes"
    Then "list-manager" should reply with dialog from "del.item.dialog"

  Scenario: Delete a list
    Given an english speaking user
    When the user says "Remove my films list"
    Then "list-manager" should reply with dialog from "confirm.deletion.dialog"
    And the user replies with "yes"
    Then "list-manager" should reply with dialog from "del.list.dialog"
```

These are currently dependent on each other which is not ideal but works as I've put them in a logical order. They'll also perform the operations on any host that runs it, so might better to at least rename the list to something like "integration test list" to avoid messing with any actual lists on your system.

Voight Kampff (and Behave) allow you to [create custom Steps](https://mycroft-ai.gitbook.io/docs/skill-development/voight-kampff/custom-steps) for a test. So these could be things like:
```
Given there is a list called films
```

This allows you to get the system into an expected state before running the test which means that each Scenario can be run independently of the others rather than needing to be in some very specific order.

Hope that helps, and at the very least the block above would get the tests passing so we can merge the update.

lb803 commented:
Thank you so much for this!! I'll do as suggested.

devops-mycroft commented:
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/skills/PR-1541)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/skills/PR-1541/logs/skills.log), [audio.log](https://reports.mycroft.ai/skills/PR-1541/logs/audio.log), [voice.log](https://reports.mycroft.ai/skills/PR-1541/logs/voice.log), [bus.log](https://reports.mycroft.ai/skills/PR-1541/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/skills/PR-1541/logs/enclosure.log)

devops-mycroft commented:
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/skills/PR-1541)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/skills/PR-1541/logs/skills.log), [audio.log](https://reports.mycroft.ai/skills/PR-1541/logs/audio.log), [voice.log](https://reports.mycroft.ai/skills/PR-1541/logs/voice.log), [bus.log](https://reports.mycroft.ai/skills/PR-1541/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/skills/PR-1541/logs/enclosure.log)

devops-mycroft commented:
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/skills/PR-1541)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/skills/PR-1541/logs/skills.log), [audio.log](https://reports.mycroft.ai/skills/PR-1541/logs/audio.log), [voice.log](https://reports.mycroft.ai/skills/PR-1541/logs/voice.log), [bus.log](https://reports.mycroft.ai/skills/PR-1541/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/skills/PR-1541/logs/enclosure.log)

devops-mycroft commented:
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/skills/PR-1541))

krisgesling commented:
Hooray we're on the other side of the VK test abyss!!!!

Changes all look good too so merging!

------------------------

A small noticing, there are a few places you have something like:
```python
    def list_empty(self, list):
        return True if len(self.json_data.get(list)) == 0 else False
```

The conditional in this return statement evaluates to `True` or `False`, so you're kind of saying:
```python
        return True if True else False
```

This means it can be simplified to:
```python
    def list_empty(self, list):
        return len(self.json_data.get(list)) == 0
```

But if you find your version more readable / whatever, there's nothing wrong with it.

lb803 commented:
I see!!
Thanks for this last tip and for your help all along!!
